models/issues/reaction.go

```diff
@@ -105,32 +168,31 @@ func findReactions(e db.Engine, opts FindReactionsOptions) ([]*Reaction, int64,
 	return reactions, count, err
 }
 
-func createReaction(e db.Engine, opts *ReactionOptions) (*Reaction, error) {
+func createReaction(ctx context.Context, opts *ReactionOptions) (*Reaction, error) {
 	reaction := &Reaction{
 		Type:    opts.Type,
-		UserID:  opts.Doer.ID,
-		IssueID: opts.Issue.ID,
+		UserID:  opts.DoerID,
+		IssueID: opts.IssueID,
 	}
 	findOpts := FindReactionsOptions{
-		IssueID:   opts.Issue.ID,
+		IssueID:   opts.IssueID,
 		CommentID: -1, // reaction to issue only
 		Reaction:  opts.Type,
-		UserID:    opts.Doer.ID,
-	}
-	if opts.Comment != nil {
-		reaction.CommentID = opts.Comment.ID
-		findOpts.CommentID = opts.Comment.ID
+		UserID:    opts.DoerID,
 	}
 
-	existingR, _, err := findReactions(e, findOpts)
+	reaction.CommentID = opts.CommentID
+	findOpts.CommentID = opts.CommentID
```

Comment:
Move these two assignments into the `&Reaction{` and `FindReactionsOptions{` ?

And the old code `CommentID: -1, // reaction to issue only` seems strange.

Reply:
done.